### ISSUE TITLE
Getting docker.errors.ImageNotFound error

### ISSUE THREAD

**Issue by jonesnc** [tox-dev/tox-docker#79]
I'm using `Python 3.8.5`, with these packages installed:
```
tox==3.19.0
tox-docker==1.7.0
```

When I try to run `tox` with this `tox.ini` file:

```
[testenv:integration-tests]
deps = pytest
commands = py.test deluge_client_wrapper/tests
docker = deluge

[docker:deluge]
image = binhex/arch-deluge:1.3.15_18_ge050905b2-1-01
volumes =
    /Users/me/deluge/data:/data
    /Users/me/deluge/config:/config
    /etc/localtime:/etc/localtime
healthcheck_cmd = curl localhost:8112
healthcheck_timeout = 10
healthcheck_retries = 30
healthcheck_interval = 1
healthcheck_start_period = 5
environment =
    DELUGE_DAEMON_LOG_LEVEL=info
    DELUGE_WEB_LOG_LEVEL=info
    UMASK=755
    PUID=1001
    PGID=1001
ports =
    8112:8112
    58846:58846
    58946:58946
```

I get this error:

```
tox -v                                                                                                                       Wed Aug 19 23:13:34 2020
using tox.ini: /Users/me/Projects/deluge-client-wrapper/tox.ini (pid 29204)
using tox-3.19.0 from /usr/local/lib/python3.8/site-packages/tox/__init__.py (pid 29204)
GLOB sdist-make: /Users/me/Projects/deluge-client-wrapper/setup.py
[29216] /Users/me/Projects/deluge-client-wrapper$ /usr/local/opt/python@3.8/bin/python3.8 setup.py sdist --formats=zip --dist-dir /Users/me/Projects/deluge-client-wrapper/.tox/dist >.tox/log/GLOB-0.log
package .tmp/package/1/deluge-client-wrapper-0.0.1.zip links to dist/deluge-client-wrapper-0.0.1.zip (/Users/me/Projects/deluge-client-wrapper/.tox)
integration-tests cannot reuse: no previous config /Users/me/Projects/deluge-client-wrapper/.tox/integration-tests/.tox-config1
integration-tests create: /Users/me/Projects/deluge-client-wrapper/.tox/integration-tests
[29230] /Users/me/Projects/deluge-client-wrapper/.tox$ /usr/local/opt/python@3.8/bin/python3.8 -m virtualenv --no-download --python /usr/local/opt/python@3.8/bin/python3.8 integration-tests >integration-tests/log/integration-tests-0.log
integration-tests installdeps: pytest
WARNING: Discarding $PYTHONPATH from environment, to override specify PYTHONPATH in 'passenv' in your configuration.
[29239] /Users/me/Projects/deluge-client-wrapper$ /Users/me/Projects/deluge-client-wrapper/.tox/integration-tests/bin/python -m pip install pytest >.tox/integration-tests/log/integration-tests-1.log
integration-tests inst: /Users/me/Projects/deluge-client-wrapper/.tox/.tmp/package/1/deluge-client-wrapper-0.0.1.zip
write config to /Users/me/Projects/deluge-client-wrapper/.tox/integration-tests/.tox-config1 as '68e5687d74599feb3623240747cf615cbe98952279aca53684d210d2398951d1 /usr/local/opt/python@3.8/bin/python3.8\n3.19.0 0 0 0\n00000000000000000000000000000000 pytest'
WARNING: Discarding $PYTHONPATH from environment, to override specify PYTHONPATH in 'passenv' in your configuration.
[29251] /Users/me/Projects/deluge-client-wrapper$ /Users/me/Projects/deluge-client-wrapper/.tox/integration-tests/bin/python -m pip install --exists-action w .tox/.tmp/package/1/deluge-client-wrapper-0.0.1.zip >.tox/integration-tests/log/integration-tests-2.log
[29257] /Users/me/Projects/deluge-client-wrapper$ /Users/me/Projects/deluge-client-wrapper/.tox/integration-tests/bin/python -m pip freeze >.tox/integration-tests/log/integration-tests-3.log
integration-tests installed: attrs==19.3.0,deluge-client==1.9.0,deluge-client-wrapper @ file:///Users/me/Projects/deluge-client-wrapper/.tox/.tmp/package/1/deluge-client-wrapper-0.0.1.zip,iniconfig==1.0.1,more-itertools==8.4.0,packaging==20.4,pluggy==0.13.1,py==1.9.0,pyparsing==2.4.7,pytest==6.0.1,six==1.15.0,toml==0.10.1
integration-tests docker: pull 'deluge'
________________________________________________________________________________________ summary _________________________________________________________________________________________
  integration-tests: commands succeeded
  congratulations :)
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 261, in _raise_for_status
    response.raise_for_status()
  File "/usr/local/lib/python3.8/site-packages/requests/models.py", line 941, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http+docker://localhost/v1.40/images/deluge/json

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/tox_docker.py", line 200, in tox_runtest_pre
    docker.images.get(image)
  File "/usr/local/lib/python3.8/site-packages/docker/models/images.py", line 316, in get
    return self.prepare_model(self.client.api.inspect_image(name))
  File "/usr/local/lib/python3.8/site-packages/docker/utils/decorators.py", line 19, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/docker/api/image.py", line 245, in inspect_image
    return self._result(
  File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 267, in _result
    self._raise_for_status(response)
  File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 263, in _raise_for_status
    raise create_api_error_from_http_exception(e)
  File "/usr/local/lib/python3.8/site-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
docker.errors.ImageNotFound: 404 Client Error: Not Found ("no such image: deluge: No such image: deluge:latest")

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 261, in _raise_for_status
    response.raise_for_status()
  File "/usr/local/lib/python3.8/site-packages/requests/models.py", line 941, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http+docker://localhost/v1.40/images/create?fromImage=deluge

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/tox", line 8, in <module>
    sys.exit(cmdline())
  File "/usr/local/lib/python3.8/site-packages/tox/session/__init__.py", line 44, in cmdline
    main(args)
  File "/usr/local/lib/python3.8/site-packages/tox/session/__init__.py", line 69, in main
    exit_code = session.runcommand()
  File "/usr/local/lib/python3.8/site-packages/tox/session/__init__.py", line 196, in runcommand
    return self.subcommand_test()
  File "/usr/local/lib/python3.8/site-packages/tox/session/__init__.py", line 224, in subcommand_test
    run_sequential(self.config, self.venv_dict)
  File "/usr/local/lib/python3.8/site-packages/tox/session/commands/run/sequential.py", line 22, in run_sequential
    runtestenv(venv, config)
  File "/usr/local/lib/python3.8/site-packages/tox/session/commands/run/sequential.py", line 73, in runtestenv
    config.pluginmanager.hook.tox_runtest_pre(venv=venv)
  File "/usr/local/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/usr/local/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/local/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/usr/local/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/usr/local/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/usr/local/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/lib/python3.8/site-packages/tox_docker.py", line 204, in tox_runtest_pre
    docker.images.pull(name, tag=tag or None)
  File "/usr/local/lib/python3.8/site-packages/docker/models/images.py", line 444, in pull
    pull_log = self.client.api.pull(
  File "/usr/local/lib/python3.8/site-packages/docker/api/image.py", line 415, in pull
    self._raise_for_status(response)
  File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 263, in _raise_for_status
    raise create_api_error_from_http_exception(e)
  File "/usr/local/lib/python3.8/site-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
docker.errors.ImageNotFound: 404 Client Error: Not Found ("pull access denied for deluge, repository does not exist or may require 'docker login': denied: requested access to the resource is denied")
```

I'm authenticated with docker hub. I've tried prepending `docker.io` to the image, but that didn't work.

I think it's using `deluge`, from the `docker = deluge` setting as the image name, not the value of `image`, but I'm not sure.

What am I doing wrong?

**Comment by dcrosta** [comment]
Looks like you're using the new tox-docker 2.x config syntax (which is what's documented at tox-docker.readthedocs.io); I haven't officially released 2.x yet, as I was waiting on one small thing, and then I got distracted by work; but I would appreciate if you have a chance to test it and report any issues you find. You can get the 2.x pre-relase with `python3 -m pip install --pre tox-docker`.

**Comment by jonesnc** [comment]
Thanks for the prompt response!

I just updated to `tox-docker==2.0.0a3` using the `pip` command you provided, and now I'm getting this error:

```
Traceback (most recent call last):
  File "/usr/local/bin/tox", line 8, in <module>
    sys.exit(cmdline())
  File "/usr/local/lib/python3.8/site-packages/tox/session/__init__.py", line 44, in cmdline
    main(args)
  File "/usr/local/lib/python3.8/site-packages/tox/session/__init__.py", line 65, in main
    config = load_config(args)
  File "/usr/local/lib/python3.8/site-packages/tox/session/__init__.py", line 81, in load_config
    config = parseconfig(args)
  File "/usr/local/lib/python3.8/site-packages/tox/config/__init__.py", line 278, in parseconfig
    pm.hook.tox_configure(config=config)  # post process config object
  File "/usr/local/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/usr/local/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/local/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/usr/local/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/usr/local/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/usr/local/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/lib/python3.8/site-packages/tox_docker/__init__.py", line 177, in tox_configure
    container_configs[container_name]["mounts"] = [
  File "/usr/local/lib/python3.8/site-packages/tox_docker/__init__.py", line 178, in <listcomp>
    _validate_volume_line(volume_line)
  File "/usr/local/lib/python3.8/site-packages/tox_docker/__init__.py", line 211, in _validate_volume_line
    raise ValueError(f"Volume {volume_line!r} is malformed")
ValueError: Volume '/Users/nathanjones/deluge/data:/data' is malformed
```

**Comment by jonesnc** [comment]
If this new error should be a separate issue, let me know and I'll create a new one.

**Comment by jonesnc** [comment]
I finally fixed that last error I mentioned in my previous comment by performing a factory reset of my Docker Desktop for Mac installation. For others reading this on Mac, what I did was:

Open the Troubleshoot window (found in the top menu dropdown) > Reset to factory defaults

Once I did that, I re-launched Docker Desktop and I no longer got the error.

Closing this issue now.

Thanks to @dcrosta for all of your help!
